frontend/src/pages/checkInPopup.tsx

```diff
@@ -0,0 +1,229 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { useEffect, useState } from "react";
+import { Dimensions, Modal, Pressable, StyleSheet, Text, View } from "react-native";
+
+import FaceIcon from "@/assets/mood-illustration.svg";
+import { lightModeColors } from "@/constants/colors";
+import { logMood } from "@/lib/api";
+
+// Update color map to match mood values
+const moodColorMap = {
+  Happy: lightModeColors.moodAccent,
+  Good: lightModeColors.moodGood,
+  Okay: lightModeColors.moodOkay,
+  Meh: lightModeColors.moodMeh,
+  Bad: lightModeColors.moodBad,
+};
+
+const moods = [
+  { label: "Happy", value: "Happy" },
+  { label: "Good", value: "Good" },
+  { label: "Okay", value: "Okay" },
+  { label: "Meh", value: "Meh" },
+  { label: "Bad", value: "Bad" },
+] as const;
+
+type MoodValue = (typeof moods)[number]["value"];
+
+const { width: screenWidth } = Dimensions.get("window");
+
+type CheckInPopupProps = {
+  userId: string;
+  onMoodLogged?: () => void;
+};
+
+export default function CheckInPopup({ userId, onMoodLogged }: CheckInPopupProps) {
+  const [showPopup, setShowPopup] = useState(false);
+  const [selectedMood, setSelectedMood] = useState<MoodValue | null>(null);
+  const [showConfirmation, setShowConfirmation] = useState(false);
+
+  useEffect(() => {
+    const checkMoodStatus = async () => {
+      const today = new Date().toISOString().split("T")[0];
+      const key = `moodCheckin-${userId}-${today}`;
+      const hasCheckedIn = await AsyncStorage.getItem(key);
+
+      if (!hasCheckedIn) {
+        setShowPopup(true);
+      }
+    };
+
+    void checkMoodStatus();
+  }, [userId]);
+
+  const handleCheckIn = async () => {
+    if (!selectedMood) return;
+
+    try {
+      // Log the mood to the backend - no need to convert case since values match
+      await logMood(userId, selectedMood);
+
+      onMoodLogged?.();
+
+      // Store in AsyncStorage to prevent multiple check-ins
+      const today = new Date().toISOString().split("T")[0];
+      const key = `moodCheckin-${userId}-${today}`;
+      await AsyncStorage.setItem(key, selectedMood);
```

Comment:
Can you remove the usage of `AsyncStorage` for logging the mood and checking whether the user has already logged their mood today? Since we already store that info on the backend, we should fetch from the backend for it. I'm wondering if the date parsing logic here is what was causing the check-ins not corresponding to the correct day @HarshGurnani